README-PRE-5.md

```diff
@@ -0,0 +1,374 @@
+
+```js
+const Api = require('kubernetes-client');
+const api = new Api.Api({ url: 'http://my-k8s-api-server.com' });
+
+const manifest0 = {
+  kind: 'Deployment',
+  apiVersion: 'extensions/v1beta1'
+  ...
+};
+const manifest1 = {
+  kind: 'ReplicationController',
+  apiVersion: 'v1'
+  ...
+};
+
+api.group(manifest0).ns('default').kind(manifest0).post({ body: manifest0 }, print);
+api.group(manifest1).ns('default').kind(manifest1).post({ body: manifest1 }, print);
+```
+
+### Object name aliases
+
+kubernetes-client supports the same aliases as
+[`kubectl`](http://kubernetes.io/docs/user-guide/kubectl-overview/)
+(*e.g.*, `ns` for `namespaces`) and the singular versions of the
+resource name (*e.g.*, `namespace` for `namespaces`). We can shorten
+the example above:
+
+```js
+core.ns('default').rc('http-rc').get(print);
+```
+
+### Switching namespaces
+
+You can call the `namespace` object to specify the namespace:
+
+```js
+core.ns('other-project').rc('http-rc').get(print);
+```
+
+### Query parameters
+
+You can optionally specify query string object `qs` to GET
+endpoints. kubernetes-client passes `qs` directly to
+[`request`](https://www.npmjs.com/package/request#requestoptions-callback).
+For example to filter based on [label
+selector](http://kubernetes.io/docs/user-guide/labels/):
+
+```js
+core.ns('default').rc.get({ qs: { labelSelector: 'service=http,component=api' } }, print);
+```
+
+### Label selector filtering
+
+kubernetes-client has a shortcut, `matchLabels`, for filtering on label
+selector equality:
+
+```js
+core.ns('default').rc.matchLabels({ service: 'http' }).get(print);
+```
+
+and a more general `match` method based on Kubernetes Match Expressions:
+
+```js
+core.ns('default').rc.match([{
+  key: 'service',
+  operator: 'In',
+  values: ['http']
+}, {
+  key: 'deploy',
+  operator: 'NotIn',
+  values: ['production', 'staging']
+}]).get(print);
+```
+
+**Note:** The Kubernetes API supports label selector filtering [for GET
+methods
+only](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/). The
+Kubernetes API does not support label selector filtering for POST,
+PUT, PATCH, or DELETE.
+
+### ThirdPartyResources
+
+You can extend the Kubernetes API using a
+[ThirdPartyResource](https://kubernetes.io/docs/user-guide/thirdpartyresources/)
+and kubernetes-client:
+
+```js
+const newResource = {
+  apiVersion: 'extensions/v1beta1',
+  kind: 'ThirdPartyResource',
+  metadata: {
+    name: 'new-resource.kubernetes-client.io'
+  },
+  description: 'Example resource',
+  versions: [{
+    name: 'v1'
+  }]
+};
+
+ext.thirdpartyresources.post({ body: newResource }, print);
+```
+
+and then extend an `ThirdPartyResource` API client with your new resources:
+
+```js
+const thirdPartyResources = new Api.ThirdPartyResources({
+  url: 'http://my-k8s-api-server.com',
+  group: 'kubernetes-client.io',
+  resources: ['customresources']  // Notice pluralization!
```

Comment:
nit: `customresources` => `customResources`

Reply:
the lower case is correct for the (deprecated) third part resources.

Reply:
:(